kratos/geometries/nurbs_curve_tessellation.h

```diff
@@ -0,0 +1,227 @@
+class NurbsCurveTessellation
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Geometry<typename TContainerPointType::value_type> GeometryType;
+    typedef NurbsCurveGeometry<TWorkingSpaceDimension, TContainerPointType> NurbsCurveGeometryType;
+    typedef typename GeometryType::IndexType IndexType;
+    typedef typename GeometryType::SizeType SizeType;
+
+private:    // static methods
+    static double DistanceToLine(
+        const typename GeometryType::CoordinatesArrayType& Point, 
+        const typename GeometryType::CoordinatesArrayType& LineA,
+        const typename GeometryType::CoordinatesArrayType& LineB
+        )
+    {
+        typename GeometryType::CoordinatesArrayType vector_v = LineA - Point;
+        typename GeometryType::CoordinatesArrayType vector_u = LineB - LineA;
+
+        return MathUtils<double>::Norm(MathUtils<double>::CrossProduct(vector_v, vector_u)) / MathUtils<double>::Norm(vector_u);
+    }
+
+public:
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Conctructor for tessellation of a nurbs curve
+    NurbsCurveTessellation()
+    {
+    }
+
+    /** 
+    * @brief This method tessellates a curve and stores the tessellation in the class
+    * From ANurbs library (https://github.com/oberbichler/ANurbs)
+    * @param pGeometry Pointer to the geometry
+    * @param PolynomialDegree The polynomial degree of the curve
+    * @param DomainInterval The curve interval which is to be tessellated
+    * @param KnotSpanIntervals The knot span intervals laying in the DomainInterval
+    * @param Tolerance Tolerance for the choral error
+    * @see ComputeTessellation
+    */
+    void Tessellate(
+        typename GeometryType::Pointer pGeometry, 
+        int PolynomialDegree,
+        NurbsInterval DomainInterval,
+        std::vector<NurbsInterval> KnotSpanIntervals,
+        const double Tolerance)
+    {
+        mTesselation = ComputeTessellation<TWorkingSpaceDimension>(
+            pGeometry,
+            PolynomialDegree,
+            DomainInterval,
+            KnotSpanIntervals,
+            Tolerance);
+    }
+
+    /** 
+    * @brief This method returns the tessellation of a curve
+    * From ANurbs library (https://github.com/oberbichler/ANurbs)
+    * @param pGeometry Pointer to the geometry
+    * @param PolynomialDegree The polynomial degree of the curve
+    * @param DomainInterval The curve interval which is to be tessellated
+    * @param KnotSpanIntervals The knot span intervals laying in the DomainInterval
+    * @param Tolerance Tolerance for the choral error
+    * @return std::map<double, typename GeometryType::CoordinatesArrayType> with the coordinates in local and working space
+    * @see ANurbs library (https://github.com/oberbichler/ANurbs)
+    */
+	static std::vector<std::pair<double, Vector>> ComputeTessellation(
+        typename GeometryType::Pointer pGeometry,
+        int PolynomialDegree,
+        NurbsInterval DomainInterval,
```

Comment:
```suggestion
        const NurbsInterval DomainInterval,
```

also, by ref or by value? (not sure how expensive this is)

Reply:
You love const, don't you :D